integr8ly/keycloak-operator
# client roles being removed and re-added on every reconcile

## Bug details
Modifying the slices inside the iteration on the slices was causing funniness. I've refactored this whole blob to make it more readable / maintainable.

## Verification
Hard to verify this really, here's the best I can do:
1. `oc new-project sso-test`
1. `oc create -f deploy/examples/keycloak_min.json`
1. `oc create -f deploy/examples/keycloak-realm-client-roles.yaml`
1. `make code/run NAMESPACE=sso-test`
1. wait for everything to come up
1. in the keycloak UI load the admin users rolemappings
1. in the client roles drop down select `realm-management`
1. see that view-realm is present
1. reload the page and repeat steps 7 and 8.
1. repeat steps 7-9 every ten seconds or so, for a few minutes to see that the `view-realm` role doesn't "toggle"
